### PR TITLE
Eating deep fried inedible items moves them to stomach

### PR DESCRIPTION
**Pull request by cacogen**
## About The Pull Request

- Having inedibles in stomach increases disgust at random intervals based on the amount
- Vomiting empties stomach contents
- Can squeeze out contents of stomachs in hand
- Stomachs only hold a combined weight class of 15 (so 5 normal sized objects)
- Updates toy singulo suicide to use stomach instead of cavity implants

## Why It's Good For The Game

Stops people using the publicly available deep fryers to brick or delete items by instead moving the item to the eater's stomach which is a semi-difficult place to retrieve them from and leads to all sorts of wacky hijinks trying to do it. Randomly vomiting up a slew of items is funny too.

## Changelog

:cl:
balance: Deep fried inedible items move to stomach when eaten instead of being deleted 
add: Having inedibles in stomach increases disgust and/or damages stomach at random intervals based on the amount
add: Vomiting empties stomach contents
add: Can squeeze out contents of stomachs in hand
expansion: Toy singulo suicide uses stomach instead of cavity implants
/:cl:

### CODE REVIEW

**Review by Zonespace27 — COMMENTED** [review]
Length over .len

File: code/modules/surgery/organs/stomach/_stomach.dm [hunk 1]
```diff
@@ -205,6 +217,18 @@
 /obj/item/organ/stomach/get_availability(datum/species/owner_species)
 	return !(NOSTOMACH in owner_species.inherent_traits)
 
+/obj/item/organ/stomach/proc/handle_contents(mob/living/carbon/human/mule, delta_time)
+	var/inedibles = 0
```
Comment (MrDoomBringer) [inline code/modules/surgery/organs/stomach/_stomach.dm:221]:
```suggestion
	var/inedibles = 0
	if(is_clown_job(mule.mind?.assigned_role)))
		return
```
through years of classical stage training, clowns have learned how to handle eating inedible things

**Review by MrDoomBringer — CHANGES_REQUESTED** [review]
Quick deep-fry the disk and feed it to the clown

**Comment by Iamgoofball** [comment]
:-1: 
massive deep fryer nerf

**Comment by cacogen** [comment]
I'm not sure how this got made about vore fetishism but there seems to be a fundamental misunderstanding of what that actually is here.

File: code/modules/surgery/organs/stomach/_stomach.dm (hunk 1 above)
Comment (cacogen) [reply code/modules/surgery/organs/stomach/_stomach.dm:221]: You can't deep fry the disk. Although you could if eating it wouldn't destroy it.

**Comment by moocowswag** [comment]
>Having inedibles in stomach increases disgust at random intervals based on the amount

Does it take the size of the item into account or just how many items there are or both.

**Comment by MrDoomBringer** [comment]
> Does it take the size of the item into account 

no

**Comment by IndieanaJones** [comment]
The security came inside and found the captain, dead with a HUGE belly full of MMIs.

**Comment by cacogen** [comment]
Yeah, that's pretty funny though. You shouldn't be able to eat mobs in the first place (at least not ones with minds) but I wouldn't take the opportunity to do that away from people.

**Comment by ATH1909** [comment]
sad

**Comment by Farquaar** [comment]
[Squeaky Boots Energy](https://youtu.be/7bMhiSQMY1M)

**Comment by itseasytosee** [comment]
> sad
> 
> now people will be lynched for trying to serve deep fried grenades or w/e

They will never expect the deep-fried fedora with a grenade inside.

**Comment by Bawhoppen** [comment]
👍 x 💯

File: code/game/objects/items/food/bread.dm [hunk 2]
```diff
@@ -325,6 +325,18 @@
 
 /obj/item/food/deepfryholder/proc/On_Consume(eater, feeder)
 	if(contents)
+		if(isliving(eater))
```
Comment (tralezab) [inline code/game/objects/items/food/bread.dm:328]: early return both of these (second check should `QDEL_LIST(contents)` before returning)

File: code/game/objects/items/food/bread.dm [hunk 3]
```diff
@@ -325,6 +325,18 @@
...
+			var/mob/living/guy_fieri = eater
+			var/obj/item/organ/stomach/stomach = guy_fieri.getorganslot(ORGAN_SLOT_STOMACH)
+			if(stomach)
```
Comment (tralezab) [inline code/game/objects/items/food/bread.dm:331]: early return this as well

File: code/modules/surgery/organs/stomach/_stomach.dm [hunk 4]
```diff
@@ -39,6 +40,16 @@
 	else
 		reagents.flags |= REAGENT_HOLDER_ALIVE
 
+/obj/item/organ/stomach/attack_self(mob/user, modifiers)
+	. = ..()
+
+	if(length(contents))
```
Comment (tralezab) [inline code/modules/surgery/organs/stomach/_stomach.dm:46]: early return

File: code/modules/surgery/organs/stomach/_stomach.dm [hunk 5]
```diff
@@ -39,6 +40,16 @@
...
+		to_chat(user, span_notice("You begin squeezing out the contents of [src]..."))
+		if(do_after(user, length(contents) SECONDS / 2))
```
Comment (tralezab) [inline code/modules/surgery/organs/stomach/_stomach.dm:48]: early return

File: code/modules/surgery/organs/stomach/_stomach.dm [hunk 6]
```diff
@@ -205,6 +218,25 @@
 /obj/item/organ/stomach/get_availability(datum/species/owner_species)
 	return !(NOSTOMACH in owner_species.inherent_traits)
 
+/obj/item/organ/stomach/proc/handle_contents(mob/living/carbon/human/mule, delta_time)
+	var/inedibles = 0
+	for(var/atom/movable/thing in contents)
+		if(!IsEdible(thing))
```
Comment (tralezab) [inline code/modules/surgery/organs/stomach/_stomach.dm:224]: early continue. if it's edible, move onto the next thing.

**Review by tralezab — CHANGES_REQUESTED** [review]
your early return game my friend, it must be practiced🙏

**Comment by tralezab** [comment]
otherwise cool brohan

**Comment by ATH1909** [comment]
how will this interact with the toy singulo suicide?

**Comment by optimumtact** [comment]
What happens if they have no stomach organ or their species takes no disgust?

you need a storage cap and an auto vomit at some level of items.

**Comment by cacogen** [comment]
I don't think you can eat without a stomach but if you manage it, it just deletes the contents of the deepfry holder like it used to. I've added auto vomit after a certain (low) combined volume of objects too now.

>how will this interact with the toy singulo suicide?

I've updated the suicide to avoid its regurgitation being doubled.

**Comment by cacogen** [comment]
@tralezab 
plz help

**Pull request closed without merging**